src/main/webapp/app/programming-submission/programming-submission.service.ts

```diff
@@ -291,12 +286,12 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
         this.exerciseBuildStateSubjects[exerciseId] = new BehaviorSubject<ExerciseBuildState | undefined>(undefined);
         this.fetchLatestPendingSubmissionByExerciseId(exerciseId)
             .pipe(
-                map(submissions => {
-                    return Object.entries(submissions).map(([participationId, submission]) => [parseInt(participationId, 10), submission]);
-                }),
+                map(Object.entries),
```

Comment:
I see that this approach was already used before, but could we also pass a domain object here to make it easier to understand?

Reply:
Looking at this again today, you are right. I refactored the code to always use the `ProgrammingSubmissionStateObj` type.

About the use of `Object.entries` here: It's not possible here for the server to already submit the domain object, as the server doesn't check the submissionDates of the latest submissions (if it's older than 2 minutes, we consider it failed) so it doesn't return the state of submission, just the latest submission if it has no result yet.